src/App.tsx

```diff
@@ -1,156 +1,429 @@
+
+      setInputValue(resetedTitle.title);
+
+      try {
+        await client.post('/todos', updatedObjectTitle);
+      } catch (error) {
+        console.error('Error occurred:', error);
+      }
+    }
+
+    if (event.key === 'Enter' && inputValue.trim() === '') {
+      setIsAddingTodoAllowed(true);
+      setErrorMessageField(true);
+    }
+  };
+
+  const deleteTodo = async (id: number) => {
+    const newTodos = todo.filter((element) => {
+      return element.id !== id;
+    });
+
+    if (newTodos === todo) {
+      setIsDeleteingTodoAllowed(true);
+      setErrorMessageField(true);
+    }
+
+    setTodo(newTodos);
+
+    try {
+      await client.delete(`/todos/${id}`);
+    } catch (error) {
+      console.error('Error occurred:', error);
+    }
+  };
+
+  const changeAll = () => {
+    const chnagedArr = todo.map((element) => {
+      return {
+        ...element,
+        completed: !element.completed,
+      };
+    });
+
+    setTodo(chnagedArr);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+  };
+
+  const searchTodo = async (id: number) => {
+    const updatedTodo = todo.map((obj) => {
+      if (obj.id === id) {
+        return {
+          ...obj,
+          completed: !obj.completed,
+        };
+      }
+
+      return obj;
+    });
+
+    const none = todo.some((element) => {
+      return element.id === id;
+    });
+
+    if (!none) {
+      setIsEditingTodoAllowed(true);
+      setErrorMessageField(true);
+    }
+
+    setTodo(updatedTodo);
+
+    try {
+      const todoToUpdate = todo.find((elem) => elem.id === id);
+
+      if (todoToUpdate) {
+        await client.patch(`/todos/${id}`, { completed: !todoToUpdate.completed });
+      }
+    } catch (error) {
+      console.error('Error occurred:', error);
+    }
+  };
+
+  const resetEverything = () => {
+    const onlyActives = todo.filter((obj) => {
+      return obj.completed === false;
+    });
+
+    setTodo(onlyActives);
+  };
+
+  useEffect(() => {
+    async function getTodo() {
+      try {
+        setIsLoading(true);
+
+        const fetchedData = await getTodos(10377);
+
+        setTodo(fetchedData);
+      } catch (error) {
+        console.error('Error occurred:', error);
+        throw error;
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    const isActive = todo.some((obj) => obj.completed === false);
+    const isFalse = todo.some((obj) => obj.completed === true);
+    const todoLength = todo.filter((obj) => {
+      return obj.completed === false;
+    });
+
+    setIsThereActiveTodo(isActive);
+    setIsThereCompletedTodos(isFalse);
+    setNumberOfActivTodos(todoLength.length);
+
+    getTodo();
+  }, [newTodo, todo]);
+
+  const visibleTodos: Todo[] = useMemo(() => todo.filter((element) => {
+    switch (selectedTab) {
+      case 'Completed':
+        return element.completed === true;
+      case 'Active':
+        return element.completed === false;
```

Comment:
Use logical opeartors 
```suggestion
        return element.completed;
      case 'Active':
        return !element.completed;
```